include/geolib/Shape.h

```diff
@@ -47,6 +50,12 @@ class Shape {
 
     Mesh mesh_;
 
+private:
+
+    mutable bool  bounding_box_cache_valid_;
+    mutable Vector3 bounding_box_min_cache_;
+    mutable Vector3 bounding_box_max_cache_;
```

Comment:
All variables lack documentation. `mutable` is also quite debatable, please reconsider that after solving the consistency problem.